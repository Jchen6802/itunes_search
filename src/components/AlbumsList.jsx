import { connect } from 'react-redux';
import React, { Component } from 'react';
import styled from 'styled-components';

import Album from './Album.jsx';
import { getAlbums } from '../utils/utils';
import { albumsSelector } from '../selectors';

const ResultsDiv = styled.div`
    margin-top: 20px;
`;

export class AlbumsList extends Component {
    render() {
        return <div>
            {this.renderResultsCount()}
            {getAlbums(this.props.albums, Album)}
        </div>;
    }

    renderResultsCount() {
        const albumsCount = this.props.albums.length;
        return albumsCount > 0 && <ResultsDiv>
            {albumsCount} results found.
        </ResultsDiv>;
    }
}

const mapStateToProps = (state) => {
    return {
        albums: albumsSelector(state)
    };
}

export default connect(mapStateToProps)(AlbumsList);
