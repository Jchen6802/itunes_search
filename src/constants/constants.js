import Immutable from 'immutable';

export const SET_ALBUMS = 'SET_ALBUMS';
export const SET_STATUS = 'SET_STATUS';
export const UPDATE_SEARCH_TERM = 'UPDATE_SEARCH_TERM';
export const FETCH_SOURCE = 'https://itunes.apple.com/search';
export const STATUSES = {
    ERROR: 'ERROR',
    INITIAL: 'INITIAL',
    SEARCHING: 'SEARCHING',
};

export const INITIAL_STATE = {
    albums: Immutable.List(),
    status: STATUSES.INITIAL,
    searchTerm: ''
};
