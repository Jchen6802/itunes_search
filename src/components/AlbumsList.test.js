import React from 'react';
import renderer from 'react-test-renderer';

import { STATUSES } from '../constants/constants';
import { AlbumsList } from './AlbumsList.jsx';

it('will initially render nothing', () => {
    const props = {
        albums: [],
        status: STATUSES.INITIAL
    };
    const albumsList = renderer.create(
        <AlbumsList {...props}/>
    ).toJSON();

    expect(albumsList).toMatchSnapshot();
});

it('will render 0 results when there are no albums after searching', () => {
    const props = {
        albums: [],
        status: STATUSES.SEARCHED
    };
    const albumsList = renderer.create(
        <AlbumsList {...props}/>
    ).toJSON();

    expect(albumsList).toMatchSnapshot();
});

it('will render the correct result count and albums after searching', () => {
    const props = {
        albums: [{
            artistName: 'someArtist',
            collectionName: 'someAlbum',
            artworkUrl100: 'albumCoverUrl'
        }],
        status: STATUSES.SEARCHED
    };
    const albumsList = renderer.create(
        <AlbumsList {...props}/>
    ).toJSON();

    expect(albumsList).toMatchSnapshot();
});
