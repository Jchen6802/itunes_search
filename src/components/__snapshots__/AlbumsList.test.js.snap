// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will initially render nothing 1`] = `<div />`;

exports[`will render 0 results when there are no albums after searching 1`] = `
<div>
  <div
    className="sc-bxivhb jUhmZc"
  >
    0
     results found.
  </div>
</div>
`;

exports[`will render the correct result count and albums after searching 1`] = `
<div>
  <div
    className="sc-bxivhb jUhmZc"
  >
    1
     results found.
  </div>
  <div
    className="column"
  >
    <div
      className="sc-bdVaJa gfWgdN"
    >
      <img
        alt="someAlbum"
        src="albumCoverUrl"
      />
      <div
        className="sc-bwzfXH hTbHfO"
      >
        someAlbum
      </div>
      <div
        className="sc-htpNat fpkfan"
      >
        someArtist
      </div>
    </div>
  </div>
</div>
`;
